otx/algorithms/action/adapters/mmaction/models/recognizers/recognizer3d.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch import nn
+
+from mmaction.models.builder import RECOGNIZERS
+from mmaction.models.recognizers.base import BaseRecognizer
+import functools
+from collections import OrderedDict
+
+
+@RECOGNIZERS.register_module(force=True)
+class Recognizer3D(BaseRecognizer):
+    """3D recognizer model framework."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # Hooks for redirect state_dict load/save
+        self._register_state_dict_hook(self.state_dict_hook)
+        self._register_load_state_dict_pre_hook(
+            functools.partial(self.load_state_dict_pre_hook, self)
+        )
+
+    @staticmethod
+    def state_dict_hook(module, state_dict, *args, **kwargs):
+        """Redirect model as output state_dict for OTX model compatibility
+        """
+        backbone_type = type(module.backbone).__name__
+        if backbone_type not in ['MoViNetBase']:
+            return
+
+        output = OrderedDict()
+        if backbone_type == 'MoViNetBase':
+            for k, v in state_dict.items():
+                if k.startswith('cls_head'):
+                    k = k.replace('cls_head.', '')
+                else:
+                    k = k.replace('backbone.', '')
+                output[k] = v
+        return output
+
+    @staticmethod
+    def load_state_dict_pre_hook(module, state_dict, *args, **kwargs):
+        """Redirect input state_dict to model for OTX model compatibility
+        """
+        backbone_type = type(module.backbone).__name__
+        if backbone_type not in ['MoViNetBase']:
+            return
+    
+        if backbone_type == 'MoViNetBase':
+            for k in list(state_dict.keys()):
+                v = state_dict.pop(k)
+                if k.startswith('classifier'):
+                    k = k.replace('classifier', 'cls_head.classifier')
+                else:
+                    k = 'backbone.'+k
+                state_dict[k] = v
+    
```

Comment:
Below functions are duplicated with original mmaction.
If there's no difference, then the below can be removed

Reply:
Reflected, thank you for nice catch.